lib/src/analyzers/lint_analyzer/rules/rules_list/avoid_returning_widgets/config_parser.dart

```diff
@@ -2,10 +2,26 @@ part of 'avoid_returning_widgets.dart';
 
 class _ConfigParser {
   static const _ignoredNamesConfig = 'ignored-names';
+  static const _ignoredAnnotationsConfig = 'ignored-annotations';
+
+  static const _defaultIgnoredAnnotations = [
+    'FunctionalWidget',
+    'swidget',
+    'hwidget',
+  ];
 
   static Iterable<String> getIgnoredNames(Map<String, Object> config) =>
-      config.containsKey(_ignoredNamesConfig) &&
-              config[_ignoredNamesConfig] is Iterable
-          ? List<String>.from(config[_ignoredNamesConfig] as Iterable)
-          : <String>[];
+      _getIterable(config, _ignoredNamesConfig) ?? [];
+
+  static Iterable<String> getIgnoredAnnotations(Map<String, Object> config) =>
+      _getIterable(config, _ignoredAnnotationsConfig) ??
+      _defaultIgnoredAnnotations;
+
+  static Iterable<String>? _getIterable(
+    Map<String, Object> config,
+    String name,
+  ) =>
+      config[name] is Iterable
```

Comment:
Previously we had `config.containsKey(_ignoredNamesConfig) && config[_ignoredNamesConfig] is Iterable` condition here. I removed `config.containsKey(_ignoredNamesConfig)` validation since it seems to be redundant or am I missing something?

Reply:
Yeah, seems like that. I will fix all other places some time after this PR will be merged. Could you please add `TODO:` for me?

Reply:
I added a PR https://github.com/dart-code-checker/dart-code-metrics/pull/342 instead of `TODO`, I performed a "`containsKey`" search in the project and it seems I already fixed all similar (only three) cases in this new PR.